### PR TITLE
fix: update dependencies

### PR DESCRIPTION
**Pull request by MichaelDeBoey**

### CODE REVIEW

**Comment by MichaelDeBoey** [comment]
@bradzacher & @JoshuaKGoldberg Any idea why tests are failing on Node v12 only?
It seems like somehow the @typescript-eslint rules aren't recognized on that Node version for some reason 🤔

File: package.json [hunk 1]
```diff
@@ -42,29 +42,29 @@
...
+        "eslint": "~8.46.0",
+        "eslint-config-prettier": "^8.9.0",
         "globals": "^13.20.0",
         "mocha": "^9.2.2",
         "npm-run-all": "^4.1.5",
         "nyc": "^15.1.0",
         "opener": "^1.5.2",
         "rimraf": "^3.0.2",
-        "typescript": "^5.1.3"
+        "typescript": "^5.1.6"
```
Comment (ota-meshi) [inline package.json]: TypeScript v5.1 has dropped support for node v12. So it seems we should use TypeScript v5.0.

```suggestion
        "typescript": "~5.0.4"
```


https://devblogs.microsoft.com/typescript/announcing-typescript-5-1/#breaking-changes

**Review by ota-meshi — APPROVED** [review]

**Comment by bradzacher** [comment]
I would just consider dropping support for nodejs 12 and 14.
Both of them have been EOL so people shouldn't be using them anywhere any more.

**Comment by MichaelDeBoey** [comment]
> I would just consider dropping support for nodejs 12 and 14.

@bradzacher We will once ESLint drops support for them (we have the same compatibility with this package)

**Comment by bradzacher** [comment]
🤷 No reason you have to wait for them to do it. They'll be waiting for the next major which will probably be a while.

By holding on you're falling behind on testing other dependencies - which means you're likely not testing real user scenarios.

Eg TS 5.1 requires node 14, typescript-eslint v6 requires node 16. You might find you haven't handled some breaking changes!

**Pull request merged**